Sources/SwiftParser/Parameters.swift

```diff
@@ -96,11 +96,31 @@ extension Parser {
     let modifiers = parseParameterModifiers(isClosure: false)
     let misplacedSpecifiers = parseMisplacedSpecifiers()
 
-    var names = self.parseParameterNames()
-    let (unexpectedBeforeColon, colon) = self.expect(.colon)
+    var names: ParameterNames
 
+    let unexpectedBeforeColon: RawUnexpectedNodesSyntax?
+    let colon: RawTokenSyntax
     let type: RawTypeSyntax
 
+    // try to parse the type regardless of the presence of the preceding colon
+    // to tackle any unnamed parameter or missing colon
+    // e.g. [X], (:[X]) or (x [X])
+    let canParseType = withLookahead { $0.canParseType() && $0.at(.comma) }
```

Comment:
Could you handle `, <type>)` as well? 
```suggestion
    let canParseType = withLookahead { $0.canParseType() && $0.at(.comma, .rightParen) }
```

Reply:
Also added a test case for that. Thanks